staging/src/k8s.io/apiserver/pkg/storage/cacher.go

```diff
@@ -369,6 +369,12 @@ func (c *Cacher) Get(ctx context.Context, key string, resourceVersion string, ob
 		return err
 	}
 
+	if !c.ready.check() && getRV == 0 {
```

Comment:
Will change the order.

Can we move the change to RLock to a followup PR? We already had one Lock() before so it's not that 0 vs 1 change. I can do this tomorrow in a spearate PR.